packages/recorder/src/main.tsx

```diff
@@ -30,12 +30,14 @@ export const Main: React.FC = ({
   window.playwrightSetSources = setSources;
   window.playwrightSetPaused = setPaused;
   window.playwrightUpdateLogs = callLogs => {
-    const newLog = new Map<string, CallLog>(log);
```

Comment:
this closes over `log` from the outer scope. If `setLog` is called later, this value won't change. Before Concurrent React, this apparently wasn't a problem because the `Main` component was rerendered often enough. It became a problem with Concurrent React though. The fix is to use an updater function so we always update based on the current version of the state.

Reply:
Thanks for the great description!